Issue data 178

averym16 opened the pull request:
**Describe what you are trying to do**
<!-- Are you trying to fix an issue? Implement a feature? -->
Match shuttles to schedules

**Steps for review**
<!-- How would you like the reviewer to test your feature? -->
Ensure schedules are properly matched to schedules and cache is implemented properly.

**Issues**
<!-- Are there any existing issues? Does it close or refer to an issue -->
<!-- If close, please write `fixes #123` where 123 is the issue number. -->
<!-- If mention/refer to an issue, write `related to #123` -->
#178

**Screenshots**
<!-- For UI/logs changes -->


**Additional Information**

jmccand commented on server/worker.py:
```diff
@@ -7,6 +7,164 @@
 import os
 import logging
 from datetime import datetime, date, timedelta
+from data.schedules import Schedule
```
This looks like duplicated code, perhaps a result of improper merge conflict resolution? It shows up again about 150 lines later.

jmccand commented:
Really nice work! I think we'll be ready to merge after a few small changes.
I see that you're caching the `is_at_stop` calculations, as well. This is a great idea! When you cache these calculations, how long does it take to recompute the matching? If it's reasonably fast, we may not need to run the worker every hour and can just run it when somebody fetches the content.

averym16 commented:
I removed the check for an hour since it runs really fast now. I added a safeguard to avoid recomputing if nothing has changed, just to avoid unnecessary load.

jmccand reviewed (approved):
Great work!
